gradle/validation/rat-sources.gradle

```diff
@@ -0,0 +1,167 @@
+                // JaSpell
+                pattern(substring: "Copyright (c) 2005 Bruno Martins")
+                // d3.js
+                pattern(substring: "THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS")
+                // highlight.js
+                pattern(substring: "THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS")
+            }
+
+            // MIT-like
+            substringMatcher(licenseFamilyCategory: "MIT  ", licenseFamilyName:"Modified BSD License") {
+                // ICU license
+                pattern(substring: "Permission is hereby granted, free of charge, to any person obtaining a copy")
+            }
+
+            // Apache
+            substringMatcher(licenseFamilyCategory: "AL   ", licenseFamilyName: "Apache") {
+                pattern(substring: "Licensed to the Apache Software Foundation (ASF) under")
+                // this is the old - school one under some files
+                pattern(substring: "Licensed under the Apache License, Version 2.0 (the &quot;License&quot;)")
+            }
+
+            substringMatcher(licenseFamilyCategory: "GEN  ", licenseFamilyName: "Generated") {
+                // <!-- svg files generated by gnuplot -->
+                pattern(substring: "Produced by GNUPLOT")
+                // <!-- snowball stemmers generated by snowball compiler -->
+                pattern(substring: "This file was generated automatically by the Snowball to Java compiler")
+                // <!-- parsers generated by antlr -->
+                pattern(substring: "ANTLR GENERATED CODE")
+            }
+
+            approvedLicense(familyName: "Apache")
+            approvedLicense(familyName: "The MIT License")
+            approvedLicense(familyName: "Modified BSD License")
+            approvedLicense(familyName: "Generated")
+        }
+    }
+
+    def printUnknownFiles() {
+        def ratXml = new XmlParser().parse(xmlReport)
+        def unknownLicenses = 0
+        ratXml.resource.each { resource ->
+            if (resource.'license-approval'.@name[0] == "false") {
+                // println('Unknown license: ' + resource.@name)
+                unknownLicenses++
+            }
+        }
+        if (unknownLicenses > 0) {
+            throw new GradleException("Found " + unknownLicenses + " files with " +
+                    "unknown licenses.")
+        }
+    }
+
+    @TaskAction
+    def rat() {
+        if (!reportDir.exists()) {
+            reportDir.mkdirs()
+        }
+        def origEncoding = System.getProperty("file.encoding")
+        try {
+            System.setProperty("file.encoding", "UTF-8") //affects the output of the ant rat task
+            generateXmlReport(reportDir)
+            printUnknownFiles()
+        } finally {
+            System.setProperty("file.encoding", origEncoding)
+        }
+    }
+}
+
+class RatPlugin implements Plugin<Project> {
```

Comment:
I wouldn't go into plugin creation at this moment. You don't need a plugin to define a task - just define the task class and add tasks of that type on projects where they're applicable. You can use ant tasks from gradle directly - no need to isolate anything.